src/HealthGPS/analysis_module.cpp

```diff
@@ -560,6 +632,44 @@ void AnalysisModule::initialise_output_channels(RuntimeContext &context) {
     channels_.emplace_back("std_yld");
     channels_.emplace_back("mean_daly");
     channels_.emplace_back("std_daly");
+
+    // Since we will be performing frequent lookups, we will store the strings and indexes in a map
+    // for quick access.
+    for (size_t i = 0; i < channels_.size(); i++) {
+        channel_index_.emplace(channels_[i], i);
+    }
+}
+
+size_t AnalysisModule::calculate_index(const Person &person) const {
+    // Get the bin index for each factor
+    std::vector<size_t> bin_indices;
+    for (size_t i = 0; i < factors_to_calculate_.size(); i++) {
+        double factor_value = person.get_risk_factor_value(factors_to_calculate_[i]);
+        auto bin_index =
+            static_cast<size_t>((factor_value - factor_min_values_[i]) / factor_bin_widths_[i]);
+        bin_indices.push_back(bin_index);
+    }
+
+    // Calculate the index in the calculated_stats_ vector
+    size_t index = 0;
+    for (size_t i = 0; i < bin_indices.size() - 1; i++) {
+        size_t accumulated_bins =
+            std::accumulate(std::next(factor_bins_.cbegin(), i + 1), factor_bins_.cend(), size_t{1},
+                            std::multiplies<>());
+        index += bin_indices[i] * accumulated_bins * channels_.size();
+    }
+    index += bin_indices.back() * channels_.size();
+
+    return index;
+}
+
+size_t AnalysisModule::get_channel_index(const std::string &channel) const {
+    auto it = channel_index_.find(channel);
+    if (it == channel_index_.end()) {
+        throw std::out_of_range("Unknown channel: " + channel);
+    }
+
+    return it->second;
```

Comment:
Could this be replaced with just `channel_index_.at(channel)`?

Reply:
Yep! 😅